libraries/chain/include/graphene/chain/vesting_balance_object.hpp

```diff
@@ -171,12 +199,77 @@ namespace graphene { namespace chain {
     * @ingroup object_index
     */
    struct by_account;
+   struct by_vesting_type;
+
+namespace detail {
+
+   /**
+      Calculate a hash for account_id_type and asset_id.
+      Use object_id.hash_value function to calculate sum of hashes from account_id_type and asset_id.
+      then subducts a result from std::numeric_limits<uint64_t>::max().
+      object_id.hash_value returns uint64_t but only 48 bits are significant
+      so std::numeric_limits<uint64_t>::max() - (48 bits + 48 bits) always correct from point of natural numbers
+   */
+   inline uint64_t vbo_mfs_hash(const account_id_type& account_id, const asset_id_type& asset_id)
+   {
+      return std::numeric_limits<uint64_t>::max() - ( hash_value(account_id) + hash_value(asset_id) );
```

Comment:
`object_id.hash_value` returns `size_t`, which is not necessarily 64 bits. Also, it defers to `std::hash<uint64_t>`, which does not necessarily return its argument directly. In other words, the result does not necessarily fit into 48 bits.

Even if `hash_value` does return the instance number directly, the resulting vbo_mfs_hash will have poor entropy because both object types use serial instance numbers starting from 0. It might be better to use `account_id.instance() ^ (asset_id.instance() << 48) ^ (asset_id.instance() >> 16` for example.